changelog.d/.towncrier_template.md.jinja

```diff
@@ -0,0 +1,63 @@
+{%- if versiondata["version"] == "main" -%}
+## Unreleased Changes
+{% else -%}
+## {{ versiondata["version"] }}
+{% endif %}
+
+*{{ versiondata["date"] }}*
+
+{% for section, _ in sections.items() %}
+{% if sections[section] %}
+{% for category, val in definitions.items() if category in sections[section] -%}
+
+{{ definitions[category]['name'] }}:
+
+{% for text, change_note_refs in sections[section][category].items() %}
+
+  {%-
+    set pr_issue_numbers = change_note_refs
+    | map('lower')
+    | map('int', default=None)
+    | select('integer')
+    | map('string')
+    | list
+  -%}
+
+  {%- set arbitrary_refs = [] -%}
+  {%- set commit_refs = [] -%}
+  {%- with -%}
+    {%- set commit_ref_candidates = change_note_refs | reject('in', pr_issue_numbers) -%}
+    {%- for cf in commit_ref_candidates -%}
+      {%- if cf | length in (7, 8, 40) and cf | int(default=None, base=16) is not none -%}
+        {%- set _ = commit_refs.append(cf) -%}
+      {%- else -%}
+        {%- set _ = arbitrary_refs.append(cf) -%}
+      {%- endif -%}
+    {%- endfor -%}
+  {%- endwith -%}
+
+- {{ text }} {%
+  if pr_issue_numbers -%}
+  ({issue}`{{ pr_issue_numbers|join('`, {issue}`') }}`)
```

Comment:
I'm uneasy about dropping the "PRs and issues" prefix. Having this start with a paren enables weird sequences when the change notes end with a period.

You can see multiple sequences like `. (#` @ https://pip-tools--2203.org.readthedocs.build/en/2203/changelog/#unreleased-changes which is what I was trying to avoid by (1) placing these refs on separate lines (so weird things in notes wouldn't make it look odd), and (2) would work with both single-liners and multi-paragraph blurbs.

Having no prefix only works when somebody forgets to end their phrase with a period.

Besides, commit and arbitrary refs have the same level of importance and I think they should follow the same style.